config/main.py

```diff
@@ -1045,20 +1138,98 @@ def add(session_name, src_ip, dst_ip, dscp, ttl, gre_type, queue, policer):
 
     if queue is not None:
         session_info['queue'] = queue
-    
+
+    if src_port is not None:
+        if get_interface_naming_mode() == "alias":
+            src_port_list = []
+            for port in src_port.split(","):
+                src_port_list.append(interface_alias_to_name(port))
+            src_port=",".join(src_port_list)
+
+        session_info['src_port'] = src_port
+        if direction is None:
+            direction = "both"
+
+    if direction is not None:
+        session_info['direction'] = direction.upper()
+
     """
     For multi-npu platforms we need to program all front asic namespaces
     """
     namespaces = sonic_device_util.get_all_namespaces()
     if not namespaces['front_ns']:
         config_db = ConfigDBConnector()
         config_db.connect()
+        if validate_mirror_session_config(config_db, session_name, None, src_port, direction) is False:
+            return
+        config_db.set_entry("MIRROR_SESSION", session_name, session_info)
+    else:
```

Comment:
@abdosi could you take a quick look at the multi-NPU logic here and make sure it looks OK to you?

Reply:
Validation need to be updated so that src_port and dst_port are in same asic/namespace.
TO_DO item (not blocking for this PR)

Reply:
Ok. will take of this in next PR. How do I validate this ?